kythe/java/com/google/devtools/kythe/analyzers/java/KytheTreeScanner.java

```diff
@@ -1376,23 +1380,54 @@ private JavaNode emitDiagnostic(TreeContext ctx, String message, String details,
     return nodes;
   }
 
+  private void loadAnnotationsFile(String fullPath, FileObject file) {
+    try {
+      InputStream stream = file.openInputStream();
+      Metadata newMetadata = metadataLoaders.parseFile(fullPath, ByteStreams.toByteArray(stream));
+      if (newMetadata == null) {
+        logger.atWarning().log("Can't load metadata %s", fullPath);
+        return;
+      }
+      metadata.add(newMetadata);
+      metadataFilePaths.add(fullPath);
+    } catch (IOException ex) {
+      logger.atWarning().log("Can't read metadata for %s", fullPath);
+    }
+  }
+
+  private void loadImplicitAnnotationsFile() {
+    URI uri = filePositions.getSourceFile().toUri();
+    String name = Paths.get(uri.getPath()).getFileName().toString();
+    try {
+      String fullPath = resolveSourcePath(name + ".pb.meta");
+      if (metadataFilePaths.contains(fullPath)) {
+        return;
+      }
+      FileObject file = Iterables.getOnlyElement(fileManager.getJavaFileObjects(fullPath), null);
+      if (file == null) {
+        // This is an expected case.
```

Comment:
getJavaFileObjects() should throw or return the same number of elements passed in.  If it doesn't return exactly one element, something has gone seriously wrong and should probably throw.

Reply:
It's my understanding that [getOnlyElement](https://guava.dev/releases/20.0/api/docs/com/google/common/collect/Iterables.html#getOnlyElement-java.lang.Iterable-T-) will throw if there is more than one element.

Reply:
That's fine too; you're converting exactly one string into a FileObject.  Returning either more or fewer than one FileObject indicates a serious problem which should likely throw.

Reply:
https://guava.dev/releases/25.0-jre/api/docs/com/google/common/collect/Iterables.html#getOnlyElement-java.lang.Iterable-

Throws NoSuchElement on empty, IllegalArgumentException if more than one.

Reply:
The overload I'm using here returns null on empty.

Reply:
Because you're passing it the null to return on empty, just omit that parameter and the null check.

Reply:
Omitting the null check causes tests to fail when getJavaFileObjects returns a single null value.